sherpa/optmethods/optoptimagic.py

```diff
@@ -0,0 +1,328 @@
+from sherpa.utils.types import (OptReturn,
+                                StatFunc)
+from .optscipy import convert_bounds_to_scipy
+
+# Will be filled out programmatically at the bottom of this file
+__all__ = []
+
+
+APPLICABLE_OPTIMAGIC_MINIMIZE_KEYWORDS = [
+    'algorithm', 'algo_options', 'numdiff_options', 
+    'error_handling', 'error_penalty', 'scaling',
+    ]
+'''Keyword arguments to `optimagic.minimize` that are exposed in Sherpa.'''
+
+
+try:
+    import optimagic as om
+    from optimagic.typing import AggregationLevel
+
+    # At first sight, this looks like it should be a decorator, but instead
+    # we make it a higher-order function, because decorated functions cannot
+    # be pickled.
+    # https://pythonicthoughtssnippets.github.io/2020/08/09/PTS13-rethinking-python-decorators.html
+    def wrap_optimagic_minimize(
+                    stat: StatFunc,
+                    x0: np.ndarray,
+                    xmin: np.ndarray,
+                    xmax: np.ndarray,
+                    **kwargs) -> OptReturn:
+        """Wrap `optimagic.minimize` to the Sherpa interface.
+        """
+
+        bounds = None
+        if hasattr(kwargs['algorithm'], 'algo_info'):
+            algo_info = kwargs['algorithm'].algo_info
+        else:
+            try:
+                algorithm = getattr(om.algos, kwargs['algorithm'])
+            except AttributeError:
+                raise AttributeError('optimagic does not have an algorithm '
+                                     f'named {kwargs["algorithm"]}. ')
+            else:
+                algo_info = algorithm.algo_info
+
+        bounds = None
+        if algo_info.supports_bounds:
+            bounds = convert_bounds_to_scipy(xmin, xmax, algo_info.is_global)
+
+        match algo_info.solver_type:
+            case AggregationLevel.SCALAR:
+                def stat_wrapper(x):
+                    return stat(x)[0]
+            case AggregationLevel.LEAST_SQUARES:
+                @om.mark.least_squares
+                def stat_wrapper(x):
+                    return stat(x)[1]
+            case AggregationLevel.LIKELIHOOD:
+                @om.mark.likelihood
+                def stat_wrapper(x):
+                    return -stat(x)[1]
+            case _:
+                # As of May 2025, there is no other option in optimagic
+                # but to future-proof, raise an error if new options are added.
+                raise NotImplementedError
+        
+
+        result = om.minimize(stat_wrapper, 
+                             x0,
+                             bounds=bounds,
+                             **kwargs)
+        
+        out = (# I don't think we'll ever get None as a return value, but Mypy complains
+               # that the type declaration in optimagic is "bool | None", while we want "bool".
+               # None certainly isn't a valid success.
+               result.success if result.success else False, 
```

Comment:
How about

```suggestion
               bool(result.success), 
```

or is this too "cute"?